ansible/roles/test/tasks/bgp_speaker.yml

```diff
@@ -0,0 +1,153 @@
+  with_items:
+    - {file_name: "config_2.ini", local_ip: '{{slb_ips[1]}}', port_num: '6000'}
+  connection: local
+
+- name: Generate configuration for the third exabgp instance
+  template: src=roles/test/templates/exabgp/config.j2 dest=/tmp/exabgp/{{item.file_name}}
+  with_items:
+    - {file_name: "config_3.ini", local_ip: '{{vlan_ips[0]}}', port_num: '7000'}
+  connection: local
+
+- name: Set the prefix to be announced
+  set_fact: announce_prefix="10.10.10.10/26"
+
+- name: Generate routes to be announced
+  template: src=roles/test/templates/exabgp/routes.j2 dest=/tmp/exabgp/routes
+  with_items:
+    - {slb_ip: '{{vlan_ips[0]}}', mux_ip_1: '{{vlan_ips[1]}}', mux_ip_2: '{{vlan_ips[2]}}', port_num_1: '5000', port_num_2: '6000', port_num_3: '7000'}
+  connection: local
+
+- name: Generate start file for exabgp instances
+  template: src=roles/test/templates/exabgp/start.j2 dest=/tmp/exabgp/{{item.file_name}}
+  with_items:
+    - {file_name: 'start.sh', config_file_1: 'config_1.ini', config_file_2: 'config_2.ini', config_file_3: 'config_3.ini', phy_ip: '{{vlan_ips[0]}}', logical_ip_1: '{{slb_ips[0]}}', logical_ip_2: '{{slb_ips[1]}}'}
+  connection: local
+
+- name: Copy exabgp related files to ptf container
+  copy: src=/tmp/exabgp/ dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Copy helper files to ptf container
+  copy: src=roles/test/files/helpers/dump.py dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Copy helper files to ptf container
+  copy: src=roles/test/files/helpers/http_api.py dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Copy helper files to ptf container
+  copy: src=roles/test/files/helpers/announce_routes.py dest={{exabgp_dir}}
+  delegate_to: "{{ptf_host}}"
+
+- name: Start exabgp instances
+  shell: chmod +x /root/exabgp/start.sh  && sh /root/exabgp/start.sh
+  delegate_to: "{{ptf_host}}"
+
+- name: Announce the routes
+  shell: python {{exabgp_dir}}/announce_routes.py {{exabgp_dir}}/routes >/dev/null 2>&1 &
+  delegate_to: "{{ptf_host}}"
+
+- name: Get PID of exabgp functions
+  shell: ps aux | grep exabgp | awk '!/grep|http_api|dump/ {print $2}'
+  register: pid
+  become: true
+  failed_when: false
+  delegate_to: "{{ptf_host}}"
+
+- debug: var=pid.stdout_lines
+
+- pause:
+    seconds: 10
+    prompt: "make sure dynamic bgp neighbors appears"
+
+- name: Gather bgp facts from bgp container
+  bgp_facts:
+
+- debug: msg="{{bgp_neighbors}}"
+
+- name: Verify bgp sessions are established
+  assert: { that: "'{{ bgp_neighbors[item]['state'] }}' == 'established'" }
+  with_items: "{{ bgp_neighbors.keys() }}"
+
+- name: Verify accepted prefixes of the dynamic neighbors are correct
+  assert: { that: "'{{ bgp_neighbors[item]['accepted prefixes'] }}' == '1'" }
+  with_items: "['{{slb_ips[0].split('/')[0]}}', '{{slb_ips[1].split('/')[0]}}', '{{vlan_ips[0].split('/')[0]}}'] "
+
+
+- name: Kill exabgp instances
```

Comment:
you can use killall to kill the process by name, you do not really need the get the pid first.